shared/grpcutils/grpcutils.go

```diff
@@ -79,3 +81,16 @@ func AppendHeaders(parent context.Context, headers []string) context.Context {
 	}
 	return parent
 }
+
+// AppendCustomErrorHeader sets a CustomErrorMetadataKey gRPC header on the passed in context,
+// using the passed in error data as the header's value. The data is serialized as JSON.
+func AppendCustomErrorHeader(ctx context.Context, errorData interface{}) error {
```

Comment:
Maybe this would be more aptly named as `SetCustomErrorHeader`. Append makes me think I can call it many times, however each time will actually overwrite the previous value. (I think?)

Reply:
No, it will not overwrite the previous value.

> SetHeader sets the header metadata. When called multiple times, all the provided metadata will be merged.